regolith/helpers/l_contactshelper.py

```diff
@@ -0,0 +1,136 @@
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        contacts = []
+        if rc.date:
+            temp_dat = date_parser.parse(rc.date).date()
+        for contact in self.gtx["contacts"]:
+            if rc.name:
+                list_add = False
+                for nam in rc.name:
+                     if nam.casefold() in contact.get('name').casefold():
+                        contacts.append(contact)
+                        list_add = True
+                if list_add:
+                    continue
+            if rc.inst and (rc.inst.casefold() in contact.get('institution').casefold()):
+                contacts.append(contact)
+                continue
+            if rc.date:
+                if rc.range:
+                    temp_dict = {"begin_date":temp_dat - dateutil.relativedelta.relativedelta(months=int(rc.range)), "end_date":temp_dat + dateutil.relativedelta.relativedelta(months=int(rc.range))} 
+                else:
+                    temp_dict = {"begin_date":temp_dat - dateutil.relativedelta.relativedelta(months=2), "end_date":temp_dat + dateutil.relativedelta.relativedelta(months=2)}                
+                if is_current(temp_dict, now=temp_dat):
+                    contacts.append(contact)
+                    continue
+            if rc.notes:
+                list_add = False
+                for note in rc.notes:
+                    if isinstance(contact.get('notes'),str):
+                        if note.casefold() in contact.get('notes').casefold():
+                            contacts.append(contact)
+                            list_add = True
+                    elif isinstance(contact.get('notes'),list):
+                        for no in contact.get('notes'):
+                             if note.casefold() in no.casefold():
+                                 contacts.append(contact)
+                                 list_add = True
```

Comment:
what are you using list_add for?  I think you can make the code cleaner and shorter by getting rid of it everywhere

Reply:
Since I'm using a for loop to iterate through the argument list, calling "continue" within the for loop will not yield my desired behavior (of skipping an iteration of the outside loop that iterates through the contact list). Thus, I'm just using list_add as a temporary variable that monitors whether I need to skip the outside for loop or not once I break out of the inner for loop., cause I can't affect the outer for loop when inside the inner for loop.